eliot-service/eliot/cache_manager.py

```diff
@@ -0,0 +1,366 @@
+        if nonblocking:
+            # NOTE(willkg): Timeout of 0 should return immediately if there's nothing
+            # there
+            timeout = 0
+        else:
+            timeout = 1000
+
+        inotify = inotify_simple.INotify(nonblocking=nonblocking)
+        watch_flags = (
+            flags.ACCESS
+            | flags.CREATE
+            | flags.DELETE
+            | flags.MODIFY
+            | flags.MOVED_FROM
+            | flags.MOVED_TO
+        )
+        watch_descriptor = inotify.add_watch(str(self.cache_dir), watch_flags)
+
+        LOGGER.info("Entering loop")
+        try:
+            while True:
+                for event in inotify.read(timeout=timeout):
+                    LOGGER.debug(event)
+                    event_flags = flags.from_mask(event.mask)
+                    # Ignore changes to directories
+                    if flags.ISDIR in event_flags:
+                        continue
+
+                    path = self.cache_dir / event.name
+                    if flags.CREATE in event_flags:
+                        self.process_create(path, event)
+                    elif flags.MOVED_FROM in event_flags:
+                        self.process_moved_from(path, event)
+                    elif flags.MOVED_TO in event_flags:
+                        self.process_moved_to(path, event)
+                    elif flags.MODIFY in event_flags:
+                        self.process_modify(path, event)
+                    elif flags.DELETE in event_flags:
+                        self.process_delete(path, event)
+                    elif flags.ACCESS in event_flags:
+                        self.process_access(path, event)
+                    else:
+                        LOGGER.debug(f"ignored {path} {event}")
+
+                yield
+
+        finally:
+            inotify.rm_watch(watch_descriptor)
+
+    def run_loop(self):
+        """Run cache manager in a loop."""
+        for event in self.event_generator():
+            pass
+
+    def process_delete(self, path, event):
+        """Process a delete event."""
+        LOGGER.debug(f"delete {path}")
+        self.mark_remove(path)
+
+    def process_create(self, path, event):
+        """Process a delete event."""
+        LOGGER.debug(f"create {path}")
+        self.update_bookkeeping(path)
+
+    def process_moved_from(self, path, event):
+        """Process a moved_from event."""
+        LOGGER.debug(f"moved_from {path}")
+        self.mark_remove(path)
+
+    def process_moved_to(self, path, event):
+        """Process a moved_to event."""
+        LOGGER.debug(f"moved_to {path}")
+        self.update_bookkeeping(path)
+
+    def process_modify(self, path, event):
+        """Process a modify event."""
+        LOGGER.debug(f"modify {path}")
+        self.update_bookkeeping(path)
+
+    def process_access(self, path, event):
+        """Process an access event."""
+        LOGGER.debug(f"access {path}")
+        self.mark_access(path)
```

Comment:
These `process_*` methods will get more complex when we support watching directory trees, so it's best to keep the structure this way even if it's repetitive.